File: [第三方] 读写Excel演示脚本.js
/**
 * 小瓶RPA演示demo，具体api请查看*流程开发文档*
 * 官网：https://rpa.pbottle.com/
 * 流程开发文档：https://gitee.com/pbottle/pbottle-rpa/wikis/pages
 */

const pbottleRPA = require('./pbottleRPA')
const fs = require('fs')
const os = require('os')
if (!fs.existsSync('./node_modules/node-xlsx')) {
    pbottleRPA.showMsg('Excel模块未安装',"请先运行 “[第三方模块安装].bat” 安装模块 ")
    pbottleRPA.tts("请先双击运行 [第三方模块安装]  安装所需模块 ")
    pbottleRPA.exit('错误：excel模块未安装' + "请先运行 “[第三方] 模块安装.bat” 安装模块 ")
}
const xlsx = require("node-xlsx");

console.log("=== Excel 读写测试 ===");
console.log(Date());

pbottleRPA.tts('Excel 读写测试')
//延迟5秒
pbottleRPA.sleep(1000*5)



pbottleRPA.tts(`将当前电脑配置信息生成EXCEL文件`)
pbottleRPA.sleep(1000*5)


//写入excel
const data = [
    ['项', '值'],
];

//加一行
data.push([
    '时间',
    Date().toString(),
])


//加一行
// console.log(pbottleRPA.getResolution())
data.push([
    '显示器分辨率',
    JSON.stringify(pbottleRPA.getResolution()),
])

//加一行
data.push([
    'CPU',
    os.cpus()[0].model,
])

const sheetOptions = {'!cols': [{wch: 15}, {wch: 50}]};  //自定义格式列宽
let buffer = xlsx.build([{name: '系统信息表', data: data}],{sheetOptions}); // Returns a buffer
fs.writeFileSync(`${__dirname}\\Excel测试表格.xlsx`,buffer);


pbottleRPA.tts(`已经生成EXCEL测试表格...请查看 `)
pbottleRPA.openDir(__dirname)
pbottleRPA.sleep(1000*5)


//读取excel
const workSheetsFromFile = xlsx.parse(`${__dirname}\\Excel测试表格.xlsx`);
const sheet = workSheetsFromFile[0].data;
console.log(sheet);

pbottleRPA.tts(`读取生成EXCEL文件内容，并写入运行日志... `)
pbottleRPA.sleep(1000*5)


pbottleRPA.tts('演示结束')
console.log('演示结束');
pbottleRPA.showMsg('演示结束','请查看运行日志')

File: python示例/[第三方] 读写Excel演示脚本.py
"""
PBottle RPA demonstration demo. Please refer to the *process development documentation* for specific APIs.
Official website: https://rpa.pbottle.com/
Process development documentation: https://rpa.pbottle.com/docs/
"""

import sys
import os
import platform
import json
import time

# Add parent directory to path to import pbottleRPA
sys.path.insert(0, os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
import pbottleRPA as rpa

# Try to import openpyxl
try:
    import openpyxl
    from openpyxl.styles import Font
except ImportError:
    rpa.showMsg(
        "Please install the third-party module first", "Run: pip install openpyxl"
    )
    rpa.tts("Please install the third-party module first. Run: pip install openpyxl")
    rpa.exit("Please install the third-party module first. Run: pip install openpyxl")


def excel_append(filename, line=None):
    """
    Append a row of data to an Excel file.

    Args:
        filename (str): Absolute file path
        line (list): Row data
    """
    if line is None:
        line = []
    print("Appending row to Excel", filename, line)

    # Read the Excel file
    workbook = openpyxl.load_workbook(filename)
    sheet = workbook.active

    # Append a new row to the end of the existing sheet
    sheet.append(line)

    # Save
    workbook.save(filename)


print("=== Excel Read/Write Test ===")
print(rpa.getTime())
rpa.tts("Excel Read/Write Test")
rpa.wait(3)
rpa.tts("Generating an Excel file with current computer configuration information")
rpa.wait(5)

# Create an Excel workbook
workbook = openpyxl.Workbook()
sheet = workbook.active
sheet.title = "pbottleRPA"

# Add header
sheet.append(["Item", "Value"])

# Make the first row bold
for cell in sheet[1]:
    cell.font = Font(bold=True)

# Set column widths
sheet.column_dimensions["A"].width = 30
sheet.column_dimensions["B"].width = 60

# Add data rows
sheet.append(
    [
        "Time",
        rpa.getTime(),
    ]
)
sheet.append(
    [
        "Display Resolution",
        json.dumps(rpa.getResolution()),
    ]
)
sheet.append(
    [
        "CPU",
        platform.processor() or "Unknown",
    ]
)

# Save the file
excel_path = os.path.join(
    os.path.dirname(os.path.dirname(os.path.abspath(__file__))),
    "Excel_test_spreadsheet.xlsx",
)
workbook.save(excel_path)
rpa.tts("Excel test spreadsheet has been generated. Please check.")
rpa.openDir(os.path.dirname(excel_path))

# Append a row of data
excel_append(excel_path, ["Item name", "New appended value"])
rpa.wait(5)

# Read Excel
workbook2 = openpyxl.load_workbook(excel_path)
sheet2 = workbook2.active

# Retrieve all data
values = []
for row in sheet2.iter_rows(values_only=True):
    values.append(row)
print(values)
rpa.tts("Excel test spreadsheet has been read and logged.")


if __name__ == "__main__":
    pass
